x-pack/plugins/apm/server/routes/alerts/chart_preview/get_transaction_duration.ts

```diff
@@ -71,15 +79,30 @@ export async function getTransactionDurationChartPreview({
         },
       },
       aggs: {
-        agg:
-          aggregationType === 'avg'
-            ? { avg: { field: transactionDurationField } }
-            : {
-                percentiles: {
-                  field: transactionDurationField,
-                  percents: [aggregationType === '95th' ? 95 : 99],
-                },
-              },
+        environment: {
+          terms: {
+            field: SERVICE_ENVIRONMENT,
+            missing: ENVIRONMENT_NOT_DEFINED.value,
+            size: 10,
+            order: {
+              [isAggTypeAvg ? 'avgLatency' : `pctLatency.${aggTypePercentile}`]:
+                'desc',
+            } as Record<string, 'desc'>,
+          },
+          aggs: {
+            ...(isAggTypeAvg
```

Comment:
What about adding a smal helper that you can also use in `x-pack/plugins/apm/server/routes/alerts/register_transaction_duration_alert_type.ts`?

```suggestion
            ...averageOrPercentileAgg({ aggregationType, transactionDurationField })
```

```ts
function averageOrPercentileAgg({ aggregationType, transactionDurationField }) {
  if (aggregationType === 'avg') {
    return { avgLatency: { avg: { field: transactionDurationField } } };
  }

  return {
    pctLatency: {
      percentiles: {
        field: transactionDurationField,
        percents: [aggregationType === '95th' ? 95 : 99],
        keyed: false as const,
      },
    },
  };
}
```